netlify/functions/manage-bot.ts

```diff
@@ -0,0 +1,363 @@
+      body: JSON.stringify({
+        error: "Bot name is required",
+      }),
+    };
+  }
+
+  if (!polygonAddress || !polygonAddress.newValue) {
+    return {
+      statusCode: 422,
+      body: JSON.stringify({
+        error: "Polygon address is required",
+      }),
+    };
+  }
+
+  if (!crew || !crew.newValue || crew.newValue.length < 1) {
+    return {
+      statusCode: 422,
+      body: JSON.stringify({
+        error: "Bot crew must have at least 1 member",
+      }),
+    };
+  }
+
+  let oldBotData: BotData;
+  try {
+    oldBotData = await checkBotAuth(botName, username);
+    delete oldBotData.imageUrl;
+  } catch (error) {
+    return {
+      statusCode: error.status,
+      body: JSON.stringify({ error: error.message }),
+    };
+  }
+
+  const paymentAddresses: PaymentAddress[] = [
+    { chain: "polygon", address: polygonAddress.newValue },
+  ];
+  if (ethereumAddress && ethereumAddress.newValue) {
+    paymentAddresses.push({
+      chain: "ethereum",
+      address: ethereumAddress.newValue,
+    });
+  }
+
+  const newBotData: BotData = {
+    ...oldBotData,
+    crew: crew.newValue,
+    paymentAddresses,
+  };
+
+  let avatarFilename = "";
+  let base64Avatar = "";
+  if (image) {
+    const decodedImage = Buffer.from(image, "base64");
+    const { data, info } = await sharp(decodedImage)
+      .resize({ width: 512 })
+      .toBuffer({ resolveWithObject: true });
+    base64Avatar = data.toString("base64");
+    avatarFilename = `${botName}.${info.format}`;
+    newBotData.image = `./avatars/${botName}.${info.format}`;
+  }
+
+  if (relegated?.newValue !== undefined) {
+    newBotData.relegated = relegated.newValue;
+  }
+
+  // Don't create a PR if the json content and image have not changed
+  if (isEqual(newBotData, oldBotData) && !image) {
+    return {
+      statusCode: 422,
+      body: JSON.stringify({ error: "Nothing changed" }),
+    };
+  }
+
+  const files: { [path: string]: string | File } = {
+    [`_data/bots/${botName}.json`]: JSON.stringify(newBotData, null, 2),
+  };
+  if (image) {
+    files[`_data/bots/avatars/${avatarFilename}`] = {
+      content: base64Avatar,
+      encoding: "base64",
+    };
+  }
+
+  const title = `Edit bot ${botName}`;
+  const body = dedent`
+  This auto-generated PR modifies the bot ${botName}
+
+  The request was made by ${username}
+  `;
+  const branchName = `bot/${botName}`;
+
+  try {
+    const res = await octokit.createPullRequest({
+      owner: process.env.GITHUB_REPO_OWNER!,
+      repo: process.env.REPO!,
+      title,
+      body,
+      head: branchName,
+      base: process.env.BRANCH_NAME,
+      changes: [
+        {
+          files,
+          commit: title,
+        },
+      ],
+    });
+
+    if (!res) {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({ error: "Failed to create pull request." }),
+      };
+    }
+
+    // submit helpdesk ticket
+    await fetch(`${process.env.URL}/.netlify/functions/request-support`, {
```

Comment:
This also could be a direct call, perhaps? But `fetch` doesn't seem *so* bad here since we're doing a bunch of other work anyway…